src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1276,48 +1253,49 @@ void SolverInterfaceImpl::writeVectorGradientData(
 }
 
 void SolverInterfaceImpl::writeBlockVectorGradientData(
-    int           dataID,
-    int           size,
-    const int *   valueIndices,
-    const double *gradientValues)
+    std::string_view meshName,
+    std::string_view dataName,
+    int              size,
+    const int *      valueIndices,
+    const double *   gradientValues)
 {
 
   PRECICE_EXPERIMENTAL_API();
 
   // Asserts and checks
-  PRECICE_TRACE(dataID, size);
+  PRECICE_TRACE(meshName, dataName, size);
   PRECICE_CHECK(_state != State::Finalized, "writeBlockVectorGradientData(...) cannot be called after finalize().");
-  PRECICE_REQUIRE_DATA_WRITE(dataID);
+  PRECICE_REQUIRE_DATA_WRITE(meshName, dataName);
   if (size == 0)
     return;
 
-  if (requiresGradientDataFor(dataID)) {
+  if (requiresGradientDataFor(meshName, dataName)) {
 
     PRECICE_CHECK(valueIndices != nullptr, "writeBlockVectorGradientData() was called with valueIndices == nullptr");
     PRECICE_CHECK(gradientValues != nullptr, "writeBlockVectorGradientData() was called with gradientValues == nullptr");
 
     // Get the data
-    WriteDataContext &context = _accessor->writeDataContext(dataID);
+    WriteDataContext &context = _accessor->writeDataContext(meshName, dataName);
     PRECICE_ASSERT(context.providedData() != nullptr);
 
-    mesh::Data &data = *context.providedData();
+    mesh::Data &meshData = *context.providedData();
 
-    // Check if the Data object with ID dataID has been initialized with gradient data
-    PRECICE_CHECK(data.hasGradient(), "Data \"{}\" has no gradient values available. Please set the gradient flag to true under the data attribute in the configuration file.", data.getName())
+    // Check if the Data object with ID mesh,data has been initialized with gradient data
```

Comment:
Comment does not make a lot of sense to me

Reply:
Thanks! Replace all is tricky
```suggestion
    // Check if the Data object of given mesh has been initialized with gradient data
```